django/test/signals.py

```diff
@@ -178,3 +178,10 @@ def static_finders_changed(**kwargs):
     }:
         from django.contrib.staticfiles.finders import get_finder
         get_finder.cache_clear()
+
+
+@receiver(setting_changed)
+def auth_user_model_changed(**kwargs):
+    if kwargs['setting'] == 'AUTH_USER_MODEL':
+        from django.apps import apps
+        apps.get_swappable_setting.cache_clear()
```

Comment:
[`test_foreign_key_swapped()`](https://github.com/django/django/blob/master/tests/field_deconstruction/tests.py#L221) and [`test_many_to_many_field_swapped()`](https://github.com/django/django/blob/master/tests/field_deconstruction/tests.py#L299) in the `field_deconstruction` test suite require this to pass.